### PR TITLE
Handle returns

### PR DESCRIPTION
**Pull request by aftersomemath**
Make MSP messaging more reliable. Handle more failure cases.

### CODE REVIEW

File: src/MspFcComms.cpp [hunk 1]
```diff
@@ -414,49 +443,103 @@ FcCommsReturns MspFcComms::receiveResponseAfterSend(
...
+                }
+                else
+                {
+                    ROS_ERROR("Partial invalid header received when receiving MSP packet");
+                    return FcCommsReturns::kReturnError;
+                }
+            }
+            else if(read_bytes > 1)
+            {
+                ROS_ASSERT("Impossible number of bytes received. Possible bug in serial library?");
```
Comment (amiller27) [inline src/MspFcComms.cpp]: This needs to be `ROS_ASSERT_MSG(false, msg)`

Comment (amiller27) [reply src/MspFcComms.cpp]: Actually because you don't have an else here, I would make this an `else` and assert that `read_bytes == 0`

Comment (amiller27) [reply src/MspFcComms.cpp]: But it doesn't really matter

**Review by amiller27 — CHANGES_REQUESTED** [review]

File: src/MspFcComms.cpp (hunk 1 above)
Comment (aftersomemath) [reply src/MspFcComms.cpp]: Adjusted

**Review by amiller27 — APPROVED** [review]

**Pull request merged**